cogs/study.py

```diff
@@ -14,43 +26,295 @@ async def pcalc(self, ctx):
+    async def handle_image(self, image_data):
+        """Takes an image dict as given by the wolf. Retrieves, trims and returns an Image object."""
+        target = image_data["src"]
+        async with aiohttp.ClientSession() as session:
+            async with session.get(target, allow_redirects=False) as resp:
+                response = await resp.read()
+        image = Image.open(BytesIO(response))
+        return self.smart_trim(image, border=10)
+
+    def smart_trim(self, im, border=0):
+        bg = Image.new(im.mode, im.size, border)
+        diff = ImageChops.difference(im, bg)
+        bbox = diff.getbbox()
+        if bbox:
+            return im.crop(bbox)
+
+    async def pods_to_filedata(self, pod_data):
+        flat_pods = await self.flatten_pods(pod_data)
+        images = await self.glue_pods(flat_pods)
+        output_data = []
+        for result in images:
+            output = BytesIO()
+            result.save(output, format="PNG")
+            output.seek(0)
+            output_data.append(output)
+        return output_data
+
+    async def pods_to_textdata(self, pod_data):
+        flat_pods = await self.flatten_pods(pod_data, text=True)
+        tabchar = "​ "
+        tab = tabchar * 2
+
+        fields = []
+        current_name = ""
+        current_lines = []
+        for title, text, level in flat_pods:
+            if level == 0:
+                if current_lines:
+                    fields.append((current_name if current_name else "Pod", "\n".join(current_lines), 0))
+                current_name = title
+                current_lines = []
+            elif title:
+                current_lines.append(f"{tab * level}**{title}**")
+            if text:
+                current_lines.append(f"{tab * (level + 1)}{text}")
+        return fields
+
+    def triage_pods(self, pod_list):
+        if "primary" in pod_list[0] and pod_list[0]["primary"]:
+            return ([pod_list[0]], pod_list[1:])
+        else:
+            important = [pod_list[0]]
+            important.extend([pod for pod in pod_list if ("primary" in pod and pod["primary"])])
+            if len(important) == 1 and len(pod_list) > 1:
+                important.append(pod_list[1])
+            extra = [pod for pod in pod_list[1:] if pod not in important]
+            return (important, extra)
+        
+    @commands.command()
+    @commands.execute("flags", flags=["text"])
+    async def cmd_query(self, ctx, *query):
+        """Get the Answer to the Query."""
+        if query == "":
+            await ctx.send(f"Please submit a valid query! Like, `{constants.COMMAND_PREFIX}ask differentiate x+y^2 with respect to x`.")
+            return
+
+        temp_msg = await ctx.reply("Sending query, please wait.")
+
+        appid = s.WOLFRAM_APPID
+
+        try:
+            result = await self.get_query(query, appid)
+        except Exception:
+            await ctx.send("An unknown exception occurred while fetching query.")
+            return
```

Comment:
This is generally not a good practice, can **any** exception occur here? 
If not you should only except those you want to except so in case some exception which we didn't realize could happen occurs here, we wouldn't know about it as it would get handled by this.